apis/iam/v1beta1/diff_test.go

```diff
@@ -0,0 +1,856 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/structuredreporting"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestIAMPolicySpecDiffers(t *testing.T) {
+	tests := []struct {
+		name    string
+		desired *IAMPolicySpec
```

Comment:
To verify the fix for the potential nil-panic, please add test cases that cover scenarios where one of `desired` or `actual` is `nil` while the other is not.